de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/progress/ConsoleProgressIndicator.java

```diff
@@ -0,0 +1,100 @@
+package de.fu_berlin.inf.dpp.server.progress;
+
+import de.fu_berlin.inf.dpp.monitoring.IProgressMonitor;
+
+import java.io.PrintStream;
+
+public class ConsoleProgressIndicator implements IProgressMonitor {
+    private PrintStream out;
+    private boolean canceled = false;
+    
+    private int total = 0;
+    private int worked = 0;
+
+    private String task;
+    private String subTask;
+
+    private int animationFrame = 0;
+
+    public ConsoleProgressIndicator(PrintStream out) {
+        this.out = out;
+    }
+
+    @Override
+    public void done() {
+        out.print('\n');
+    }
+
+    @Override
+    public void subTask(String name) {
+        if (!name.equals(subTask)) {
+            subTask = name;
+            print();
+        }
+    }
+
+    @Override
+    public void setTaskName(String name) {
+        if (!name.equals(task)) {
+            task = name;
+            print();
+        }
+    }
+
+    @Override
+    public void worked(int amount) {
+        if (amount != worked) {
+            worked = amount;
+            print();
+        }
+    }
+
+    @Override
+    public void setCanceled(boolean canceled) {
+        if (canceled != this.canceled) {
+            this.canceled = canceled;
+            print();
+        }
+    }
+
+    @Override
+    public boolean isCanceled() {
+        return canceled;
+    }
+
+    @Override
+    public void beginTask(String name, int size) {
+        task = name;
+        total = size;
+        print();
+    }
+
+    private void print() {
+        if (canceled) {
+            out.print("[CANCELED] ");
+        } else if (total == IProgressMonitor.UNKNOWN) {
+            switch (animationFrame) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Add a default case to this switch. ([squid:SwitchLastCaseIsDefaultCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=142&issues=AWSoKT6Gl6vpliCepUn6&open=AWSoKT6Gl6vpliCepUn6)

[ISSUE_KEY]: AWSoKT6Gl6vpliCepUn6